src/utils/power_flow/make_pf.jl

```diff
@@ -41,124 +31,106 @@ res = NLsolve.nlsolve(pf!, x0)
-            x0[var_count] = total_gen[2]
-            push!(internal.args, :($Q_name = x[$(var_count)])); var_count += 1
-            push!(internal.args, :(P_bal[$ix] = $P_name + $net_p_load))
-            push!(internal.args, :(Q_bal[$ix] = $Q_name + $net_q_load))
-            push!(internal.args, :(V[$ix] = $Vm_name*(cos($ang_name)+sin($ang_name)*1im)))
-            push!(internal.args, :(Vc[$ix] = conj(V[$ix])))
-            # Reference for the results Dict
-            res_dict[b.name] = [var_ref1, var_ref2]
+            x0[state_variable_count]     = P_GEN_BUS[ix]
+            x0[state_variable_count + 1] = Q_GEN_BUS[ix]
+            state_variable_count += 2
         elseif b.bustype == PV::BusType
-            Q_name = Symbol("Q_", bus_number)
-            net_p_load = :($(total_gen[1]) - $(total_load[1]))
-            net_q_load = :(-$(total_load[2]))
-            push!(internal.args, :($Vm_name = $(bus_voltage)))
-            var_ref1 = (:reactivepower, var_count)
-            x0[var_count] = total_gen[2]
-            push!(internal.args, :($Q_name = x[$(var_count)])); var_count += 1
-            var_ref2 = (:angle, var_count)
-            x0[var_count] = bus_angle
-            push!(internal.args, :($ang_name = x[$(var_count)])); var_count += 1
-            push!(internal.args, :(P_bal[$ix] = $net_p_load))
-            push!(internal.args, :(Q_bal[$ix] = $Q_name + $net_q_load))
-            push!(internal.args, :(V[$ix] = $Vm_name*(cos($ang_name)+sin($ang_name)*1im)))
-            push!(internal.args, :(Vc[$ix] = conj(V[$ix])))
-            # Reference for the results Dict
-            res_dict[b.name] = [var_ref1, var_ref2]
+            x0[state_variable_count]     = Q_GEN_BUS[ix]
+            x0[state_variable_count + 1] = bus_angle
+            state_variable_count += 2
         elseif b.bustype == PQ::BusType
-            net_p_load = :($(total_gen[1]) - $(total_load[1]))
-            net_q_load = :($(total_gen[2]) - $(total_load[2]))
-            var_ref1 = (:voltage, var_count)
-            x0[var_count] = bus_voltage
-            push!(internal.args, :($Vm_name = x[$(var_count)])); var_count += 1
-            var_ref2 = (:angle, var_count)
-            x0[var_count] = bus_angle
-            push!(internal.args, :($ang_name = x[$(var_count)])); var_count += 1
-            push!(internal.args, :(P_bal[$ix] = $net_p_load))
-            push!(internal.args, :(Q_bal[$ix] = $net_q_load))
-            push!(internal.args, :(V[$ix] = $Vm_name*(cos($ang_name)+sin($ang_name)*1im)))
-            push!(internal.args, :(Vc[$ix] = conj(V[$ix])))
-            # Reference for the results Dict
-            res_dict[b.name] = [var_ref1, var_ref2]
+            x0[state_variable_count]     = bus_voltage
+            x0[state_variable_count + 1] = bus_angle
+            state_variable_count += 2
         end
     end
 
-    balance_eqs = quote  end
+    @assert state_variable_count - 1 == N_BUS * 2
```

Comment:
`@assert state_variable_count == N_BUS * 2`

Reply:
I think the easy readability of the rest of the code above this makes up for the awkwardness of this one line. Thoughts?

Reply:
Maybe just leave a comment of why its there.